cli/mfg.c

```diff
@@ -84,8 +85,168 @@ static int ping(int argc, char **argv)
+		state->basic_setting_valid? "(Valid)":"(Invalid)");
+
+	printf("\tSecure State: \t\t\t");
+	switch(state->secure_state) {
+	case SWITCHTEC_UNINITIALIZED_UNSECURED:
+		printf("UNINITIALIZED_UNSECURED\n");
+		break;
+	case SWITCHTEC_INITIALIZED_UNSECURED:
+		printf("INITIALIZED_UNSECURED\n");
+		break;
+	case SWITCHTEC_INITIALIZED_SECURED:
+		printf("INITIALIZED_SECURED\n");
+		break;
+	default:
+		printf("Unsupported State\n");
+		break;
+	}
+
+	printf("\tJTAG/EJTAG Debug State: \t");
+	switch(state->debug_mode) {
+	case SWITCHTEC_DEBUG_MODE_ENABLED:
+		printf("Always Enabled\n");
+		break;
+	case SWITCHTEC_DEBUG_MODE_DISABLED_BUT_ENABLE_ALLOWED:
+		printf("Disabled by Default But Can Be Enabled\n");
+		break;
+	case SWITCHTEC_DEBUG_MODE_DISABLED:
+		printf("Always Disabled\n");
+		break;
+	default:
+		printf("Unsupported State\n");
+		break;
+	}
+
+	printf("\tJTAG/EJTAG State After Reset: \t%d\n",
+		state->jtag_lock_after_reset);
+
+	printf("\tJTAG/EJTAG State After BL1: \t%d\n",
+		state->jtag_lock_after_bl1);
+
+	printf("\tJTAG/EJTAG Unlock IN BL1: \t%d\n",
+		state->jtag_bl1_unlock_allowed);
+
+	printf("\tJTAG/EJTAG Unlock AFTER BL1: \t%d\n",
+		state->jtag_post_bl1_unlock_allowed);
+
+	printf("\tSPI Clock Rate: \t\t%s MHz\n",
+		spi_rate_str[state->spi_clk_rate-1]);
+
+	printf("\tI2C Recovery TMO: \t\t%d Second(s)\n",
+		state->i2c_recovery_tmo);
+	printf("\tI2C Port: \t\t\t%d\n", state->i2c_port);
+	printf("\tI2C Address (7-bits): \t\t0x%02x\n", state->i2c_addr);
+	printf("\tI2C Command Map: \t\t0x%08x\n\n", state->i2c_cmd_map);
+
+	printf("Exponent Hex Data %s: \t\t0x%08x\n",
+		state->public_key_exp_valid? "(Valid)":"(Invalid)",
+		state->public_key_exponent);
+
+	printf("KMSK Entry Number %s: \t\t%d\n",
+		state->public_key_num_valid? "(Valid)":"(Invalid)",
+		state->public_key_num);
+
+	if (state->public_key_ver)
+		printf("Current KMSK index %s: \t\t%d\n",
+			state->public_key_ver_valid? "(Valid)":"(Invalid)",
+			state->public_key_ver);
+	else
+		printf("Current KMSK index %s: \t\tNot Set\n",
+			state->public_key_ver_valid? "(Valid)":"(Invalid)");
+
+	for(key_idx = 0; key_idx < state->public_key_num; key_idx++) {
+		printf("KMSK Entry %d:  ", key_idx + 1);
+		for(i = 0; i < SWITCHTEC_KMSK_LEN; i++)
+				printf("%02x", state->public_key[key_idx][i]);
+		printf("\n");
+	}
+}
+
+static int info(int argc, char **argv)
+{
+	const char *desc = "Display security settings (BL1 and Main Firmware only)";
+	int ret;
+	enum switchtec_boot_phase phase_id;
+
+	struct switchtec_sn_ver_info sn_info = {};
+
+	static struct {
+		struct switchtec_dev *dev;
+	} cfg = {};
+
+	const struct argconfig_options opts[] = {
+		DEVICE_OPTION_NO_PAX,
+		{NULL}
+	};
+
+	struct switchtec_security_cfg_stat state = {};
+
+	argconfig_parse(argc, argv, desc, opts, &cfg, sizeof(cfg));
+
+	ret = switchtec_get_device_info(cfg.dev, &phase_id, NULL, NULL);
+	if (ret < 0) {
+		switchtec_perror("mfg info");
+		return ret;
+	}
+
+	printf("Current Boot Phase: \t\t\t");
```

Comment:
Looks like this could be made common with the code in ping()

Reply:
Good idea. Moved duplicate switch...case to phase_id_to_string().